.github/workflows/ci.yml

```diff
@@ -8,9 +8,153 @@ on:
       - '**'
 
 env:
+  ### compiler options
+  HOST:
+  WRAPPER_CMD:
+  # Specific warnings can be disabled with -Wno-error=foo.
+  # -pedantic-errors is not equivalent to -Werror=pedantic and
+  # thus not implied by -Werror according to the GCC manual.
+  WERROR_CFLAGS: '-Werror -pedantic-errors'
+  MAKEFLAGS: '-j4'
+  BUILD: 'check'
+  ### secp256k1 config
+  ECMULTWINDOW: 'auto'
+  ECMULTGENPRECISION: 'auto'
+  ASM: 'no'
+  WIDEMUL: 'auto'
+  WITH_VALGRIND: 'yes'
+  EXTRAFLAGS:
+  ### secp256k1 modules
+  EXPERIMENTAL: 'no'
+  ECDH: 'no'
+  RECOVERY: 'no'
+  SCHNORRSIG: 'no'
+  ELLSWIFT: 'no'
+  ### test options
+  SECP256K1_TEST_ITERS:
+  BENCH: 'yes'
   SECP256K1_BENCH_ITERS: 2
+  CTIMETESTS: 'yes'
+  # Compile and run the examples.
+  EXAMPLES: 'yes'
 
 jobs:
+  docker_wine_cache:
+    name: "Build MinGW/MSVC Debian image"
+    runs-on: ubuntu-latest
+    steps:
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v2
+
+      - name: Build container
+        uses: docker/build-push-action@v4
+        with:
+          file: ./ci/linux-debian-wine.Dockerfile
+          tags: ci-image
+          push: false
+          load: false
+          cache-from: type=gha
+          cache-to: type=gha,mode=min
+
+  mingw_env_vars:
+    name: "Prepare environment variables"
+    runs-on: ubuntu-latest
+    env:
+      WRAPPER_CMD: 'wine'
+      WITH_VALGRIND: 'no'
+      ECDH: 'yes'
+      RECOVERY: 'yes'
+      SCHNORRSIG: 'yes'
+      CTIMETESTS: 'no'
+    outputs:
+      json: ${{ steps.result.outputs.json }}
+    steps:
+      - name: Put environment variables into a standard job output
+        id: result
+        run: |
+          echo "json=$(echo '${{ toJSON(env) }}' | jq -c '.')" >> "$GITHUB_OUTPUT"
```

Comment:
This still seems a bit overkill to me. Couldn't we list the env variables just in the `mingw_debian` job?

What I imagine is something like this:
```
mingw_debian:
  env:
    - WRAPPER_CMD: 'wine'
```
Am I right that this doesn't work, because it actually sets these environment variables, but then it's not clear to `run-in-docker` which variables should be passed to the inside of the container?

But what about setting `common_env_vars` only inside `mingw_debian`? Something like  
```
mingw_debian:
  with:
      common_env_vars: |
        WRAPPER_CMD=wine
        [...]
```

like this, with a more consistent syntax   (using an array of size 1 for `common_configuration`)

```
mingw_debian:
  strategy:
    matrix:
      common_configuration:
        - common_env_vars:
           WRAPPER_CMD: 'wine'
           [...]
      configuration:
[...]
```

Reply:
https://docs.github.com/en/actions/using-workflows/reusing-workflows#limitations:
>  - Any environment variables set in an `env` context defined at the workflow level in the caller workflow are not propagated to the called workflow.
>
>  - Similarly, environment variables set in the env context, defined in the called workflow, are not accessible in the env context of the caller workflow. Instead, you must use outputs of the reusable workflow.

Reply:
> But what about setting `common_env_vars` only inside `mingw_debian`?

It seems working .Going to give it a try.

Reply:
> But what about setting `common_env_vars` only inside `mingw_debian`?

This works for  `common_env_vars`, but we still need a way to pass global environment variables to the called workflow or duplicate them inside `mingw_debian`.